scripts/generate-locales.ts

```diff
@@ -0,0 +1,118 @@
+import ts from 'typescript';
+import path, { relative } from 'path';
+import { writeFile, mkdir, stat } from 'fs/promises';
+import { glob, chalk } from 'zx';
+
+const SRC_DIR = path.resolve('./src/lib/i18n/locales');
+const DEST_DIR = path.resolve('./static/i18n/locales');
+
+const logAndExit = (msg: string, logError: (msg: string) => void) => {
+  logError(chalk.red(msg));
+
+  if (logError === console.error) {
+    process.exit(0);
+  }
+};
+
+const relativeTo = (path: string) => {
+  return relative('./', path);
+};
+
+export const generateLocales = async (
+  src: string = SRC_DIR,
+  dest: string = DEST_DIR,
+  logError: (msg: string) => void = console.error,
+) => {
+  const pattern = path.join(src, '*/*.ts');
+  const filesToTranslate = await glob(pattern);
+
+  filesToTranslate.map(async (source) => {
+    const program = ts.createProgram([source], ts.getDefaultCompilerOptions());
+    const sourceFile = program.getSourceFile(source);
+    const checker = program.getTypeChecker();
+
+    if (!sourceFile) {
+      logAndExit(
+        `Error retrieving source file information for ${source}`,
+        logError,
+      );
+    }
+
+    const sourceFileSymbol = checker.getSymbolAtLocation(sourceFile!);
+
+    if (!sourceFileSymbol) {
+      logAndExit(`Error parsing source file: ${source}`, logError);
+    }
+
+    const defaultExport = checker.tryGetMemberInModuleExports(
+      'default',
+      sourceFileSymbol!,
+    );
+
+    if (!defaultExport) {
+      logAndExit(`No default export in source: ${source}`, logError);
+    }
+
+    const jsObject: Record<string, string> = {};
+    if (defaultExport?.declarations && defaultExport.declarations[0]) {
+      defaultExport.declarations[0].forEachChild((child) => {
+        if (child.kind === ts.SyntaxKind.ObjectLiteralExpression) {
+          child.forEachChild((grandchild) => {
+            if (grandchild.kind === ts.SyntaxKind.PropertyAssignment) {
+              let key = '';
+              let value = '';
+              const [keyNode, valueNode] = [
```

Comment:
I don't feel strongly about this. But if you're going to pop it into an array just to destructure it and that takes four lines, maybe you just do two variable assignments?